src/util/pg.py

```diff
@@ -196,27 +196,41 @@ def explode_query_range(cr, query, table, alias=None, bucket_size=10000, prefix=
 
     alias = alias or table
 
-    cr.execute("SELECT min(id), max(id) FROM {}".format(table))
-    min_id, max_id = cr.fetchone()
-    if min_id is None:
+    cr.execute(
+        format_query(
+            cr,
+            """
+            SELECT min(id), max(id)
+              FROM (SELECT id,
+                           ROW_NUMBER() OVER (ORDER BY id) / %s
```

Comment:
`row_number` start at 1.
```suggestion
                           (ROW_NUMBER() OVER () - 1) / %s
```